### PR TITLE
formatting, moved some code to functions

### PR DESCRIPTION
**Pull request by skryzhny**

### CODE REVIEW

File: git-keeper.py [hunk 1]
```diff
@@ -24,63 +24,82 @@
...
-            aws_secret_access_key = cnf["s3"]["aws_secret_access_key"])
-    except botocore.exceptions.NoCredentialsError:
-        raise Exception('No AWS credentials found.')
-    except botocore.exceptions.ClientError:
-        raise Exception('Invalid AWS credentials.')
+    aws_access_key_id = cnf["s3"]["aws_access_key_id"]
+    aws_secret_access_key = cnf["s3"]["aws_secret_access_key"]
+    s3_bucket = cnf["s3"]["bucket"]
+    date = datetime.now().strftime('%Y-%m-%d')
+    check_aws_creds(aws_access_key_id, aws_secret_access_key)
```
Comment (jmelis) [inline git-keeper.py]: perhaps you can return the s3 creds in this function, that will allow you to not need to run the same statement again in line 91

File: git-keeper.py [hunk 2]
```diff
@@ -24,63 +24,82 @@
 tar = sh.tar.bake('-cf')
 workdir = 'workdir'
 
+
 def upload2s3(s3_client, repo_tar, git_keeper_bucket, date, object_name):
     try:
-        response = s3_client.upload_file(repo_tar, git_keeper_bucket, date + '/' + object_name)
+        s3_client.upload_file(
+            repo_tar, git_keeper_bucket, date + '/' + object_name)
```
Comment (jmelis) [inline git-keeper.py]: I still feel like the try/catch surrounding this is unnecessary. What are we gaining with this try/catch? Without the try/catch the execution will also stop if it raises a ClientError.

File: git-keeper.py [hunk 3]
```diff
@@ -24,63 +24,82 @@
...
 def cleanwrkdir(workdir):
     shutil.rmtree(workdir, ignore_errors=True)
     os.makedirs(workdir, exist_ok=True)
 
+
 def clone_repo(repo_url, repo_dir):
     mirror(repo_url, repo_dir)
 
+
+def check_aws_creds(aws_access_key_id, aws_secret_access_key):
```
Comment (jmelis) [inline git-keeper.py]: rename to get_s3_client  and return the client at the end

File: git-keeper.py [hunk 4]
```diff
@@ -24,63 +24,82 @@
...
+                     aws_access_key_id=aws_access_key_id,
+                     aws_secret_access_key=aws_secret_access_key)
+    except botocore.exceptions.NoCredentialsError:
+        raise Exception('No AWS credentials found.')
+    except botocore.exceptions.ClientError:
+        raise Exception('Invalid AWS credentials.')
+
+
+def git_clone_upload(s3_client, gpg, recipients, repo, s3_bucket, date):
+    repo_url = repo + '.git'
```
Comment (jmelis) [inline git-keeper.py]: I think this should be:

```
if not repo.endswith('.git'):
  repo += '.git'
```

File: git-keeper.py [hunk 5]
```diff
@@ -24,63 +24,82 @@
...
+    repo_dir = workdir + '/' + os.path.basename(repo_url)
```
Comment (jmelis) [inline git-keeper.py]: use `os.path.join(workdir, os.path.basename(repo_url))` instead

File: git-keeper.py (hunk 2 above)
Comment (jmelis) [reply git-keeper.py]: also, use `os.path.join` for  the last component.

File: git-keeper.py [hunk 6]
```diff
@@ -24,63 +24,82 @@
...
-            aws_access_key_id = cnf["s3"]["aws_access_key_id"],
-            aws_secret_access_key = cnf["s3"]["aws_secret_access_key"])
-    except botocore.exceptions.NoCredentialsError:
-        raise Exception('No AWS credentials found.')
-    except botocore.exceptions.ClientError:
-        raise Exception('Invalid AWS credentials.')
+    aws_access_key_id = cnf["s3"]["aws_access_key_id"]
+    aws_secret_access_key = cnf["s3"]["aws_secret_access_key"]
+    s3_bucket = cnf["s3"]["bucket"]
+    date = datetime.now().strftime('%Y-%m-%d')
```
Comment (jmelis) [inline git-keeper.py]: can you add empty new lines so it's not all grouped together, but split into logical groups?

```
# get s3 client
...

# read configuration
...

# set the date
....
```

File: git-keeper.py (hunk 1 above)
Comment (skryzhny) [reply git-keeper.py]: Yes, returning s3_client

File: git-keeper.py (hunk 2 above)
Comment (skryzhny) [reply git-keeper.py]: I think we want to log error and continue to next repo

File: git-keeper.py (hunk 4 above)
Comment (skryzhny) [reply git-keeper.py]: Sure, it's better. I just want in this PR deal with formatting and maybe moving to functions

**Pull request merged**